### ISSUE TITLE
[$60]Challenge Details: Eligible events section not displayed

### ISSUE THREAD

**Issue by SathyaJayabal** [topcoder-platform/community-app#4430]
V5 URL: https://beta-community-app.topcoder.com/challenges/d9c74629-762f-4464-9753-ebd982c7990f

V4 URL: https://www.topcoder.com/challenges/30106190
<img width="1440" alt="Screenshot 2020-05-27 at 4 49 36 PM" src="https://user-images.githubusercontent.com/58783823/83013121-1939ff00-a03a-11ea-9405-0d3a1dac5f90.png">
<img width="1440" alt="Screenshot 2020-05-27 at 4 48 20 PM" src="https://user-images.githubusercontent.com/58783823/83013146-22c36700-a03a-11ea-9c0d-525084a6c75a.png">

**Comment by SathyaJayabal** [comment]
FYI: This issue  was  tested on production backend. Please use the given urls to compare
V5 URL: https://beta-community-app.topcoder.com/challenges/d9c74629-762f-4464-9753-ebd982c7990f
V4 URL: https://www.topcoder.com/challenges/30106190

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30126891 has been created for this ticket.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30126891 has been updated - it has been assigned to sr_jr.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by simranb86** [comment]
@sushilshinde @rootelement can you help how do we get events from challenge-api, I don't see any placeholder on API response for `curl -X GET "https://api.topcoder.com/v5/challenges/:challengeId" -H "accept: application/json"` ?

**Comment by rootelement** [comment]
This has to be migrated. Working on adding this to the metadata object. it will be as such:
```json
metadata: {
  "events": [
    { "id": 1, "name": "2017 TopCoder(R) Open", "key": "tco17" }
  ]
}
```

**Comment by rootelement** [comment]
it gets id, event_desc, and event_short_desc

**Comment by rootelement** [comment]
This is done in dev:
http://api.topcoder-dev.com/v5/challenges?legacyId=30051154

Will run on prod overnight

**Comment by sushilshinde** [comment]
@simranb86 waiting on PR

**Comment by simranb86** [comment]
PRs:
- https://github.com/topcoder-platform/topcoder-react-lib/pull/176
- https://github.com/topcoder-platform/community-app/pull/4443

**Comment by simranb86** [comment]
> @simranb86 waiting on PR

Sorry I thought I already pushed it, has it ready once received the info. Pushed in PRs now.

**Comment by SathyaJayabal** [comment]
@simranb86 , Cannot  seethe eligible events section on beta
<img width="1440" alt="Screenshot 2020-06-01 at 4 33 28 PM" src="https://user-images.githubusercontent.com/58783823/83403184-d0ad8780-a425-11ea-8113-53f567092166.png">

**Comment by SathyaJayabal** [comment]
@sushilshinde , can see the  eligible events on dev, but not on prod
example: https://beta-community-app.topcoder.com/challenges/d9c74629-762f-4464-9753-ebd982c7990f
cc @rootelement 
<img width="1440" alt="Screenshot 2020-06-01 at 5 31 58 PM" src="https://user-images.githubusercontent.com/58783823/83407218-e161fb80-a42d-11ea-8186-76cf976ca0f1.png">

**Comment by simranb86** [comment]
@SathyaJayabal yes, this is data issue, plz make sure you test with challenge which has events in it - try this via postman `https://api.topcoder-dev.com/v5/challenges/<challengeId>` to see if that challenge has events there in response.

**Comment by SathyaJayabal** [comment]
@simranb86, on testing after the data migration in dev. I can see  the events in the api response, but not in the front end. Can you please take a look.
cc @sushilshinde 
https://test-community-app.topcoder-dev.com/challenges/3c56f0ad-5060-443f-990a-94c384ec41d3
http://api.topcoder-dev.com/v5/challenges/3c56f0ad-5060-443f-990a-94c384ec41d3
V4 URL:https://qa-community-app.topcoder-dev.com/challenges/30051173
<img width="1440" alt="Screenshot 2020-06-02 at 4 49 40 PM" src="https://user-images.githubusercontent.com/58783823/83514353-1f742380-a4f1-11ea-8ddd-50027dcb0d88.png">
<img width="1440" alt="Screenshot 2020-06-02 at 4 49 47 PM" src="https://user-images.githubusercontent.com/58783823/83514365-2733c800-a4f1-11ea-89fc-adf7857177c0.png">

**Comment by simranb86** [comment]
> @sushilshinde , can see the eligible events on dev, but not on prod
> example: https://beta-community-app.topcoder.com/challenges/d9c74629-762f-4464-9753-ebd982c7990f
> cc @rootelement
> <img alt="Screenshot 2020-06-01 at 5 31 58 PM" width="1440" src="https://user-images.githubusercontent.com/58783823/83407218-e161fb80-a42d-11ea-8186-76cf976ca0f1.png">

@SathyaJayabal The backend impl was changed, it is re-work to be done here. The metadata was an object but now it has been updated to be an array. 

@sushilshinde FYI ☝️

**Comment by rootelement** [comment]
I apologize @simranb86 , metadata was always intended to be an array of objects with {type, value}. 

See here:
http://api.topcoder-dev.com/v5/challenges/ce01f468-e5cc-4c03-abe6-0e625af94ec1

Events is no different:
http://api.topcoder-dev.com/v5/challenges/3c56f0ad-5060-443f-990a-94c384ec41d3

**Comment by simranb86** [comment]
> I apologize @simranb86 , metadata was always intended to be an array of objects with {type, value}.
> 
> See here:
> http://api.topcoder-dev.com/v5/challenges/ce01f468-e5cc-4c03-abe6-0e625af94ec1
> 
> Events is no different:
> http://api.topcoder-dev.com/v5/challenges/3c56f0ad-5060-443f-990a-94c384ec41d3

Thanks @rootelement 

@SathyaJayabal @sushilshinde give me like 30 mins, will raise new PR for the changes. Thanks

**Comment by sushilshinde** [comment]
@simranb86 please link PR

**Comment by simranb86** [comment]
> @simranb86 please link PR

@sushilshinde this is pending for @rootelement to provide a challenge which has events in valid format. I already flagged him on slack and waiting for him to provide one which could be used for development and then can pass to qa to validate it.

**Comment by rootelement** [comment]
Events is now a first class object:
http://api.topcoder-dev.com/v5/challenges?legacyId=30051173

Also, metadata is `{name, value}`, not `{type, value}`
http://api.topcoder-dev.com/v5/challenges?legacyId=30048648

please update accordingly and see my big update in slack for today.

**Comment by simranb86** [comment]
> Events is now a first class object:
> http://api.topcoder-dev.com/v5/challenges?legacyId=30051173
> 
> Also, metadata is `{name, value}`, not `{type, value}`
> http://api.topcoder-dev.com/v5/challenges?legacyId=30048648
> 
> please update accordingly and see my big update in slack for today.

Thanks @rootelement 

@sushilshinde PR for the same https://github.com/topcoder-platform/topcoder-react-lib/pull/180

**Comment by sushilshinde** [comment]
Thanks @simranb86

**Comment by SathyaJayabal** [comment]
verified on beta
<img width="1440" alt="Screenshot 2020-06-05 at 8 23 46 PM" src="https://user-images.githubusercontent.com/58783823/83891652-24440c00-a76b-11ea-939e-042a6bf532a3.png">
<img width="1440" alt="Screenshot 2020-06-05 at 8 24 41 PM" src="https://user-images.githubusercontent.com/58783823/83891668-2a39ed00-a76b-11ea-95e4-c83ea30a921d.png">

**Comment by crazyk07** [comment]
Payment task has been updated: https://software.topcoder.com/review/actions/ViewProjectDetails?pid=30126891<br/><br/>```This is an automated message for crazyk via Topcoder X```
